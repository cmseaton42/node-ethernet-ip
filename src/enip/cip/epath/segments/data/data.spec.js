const { build } = require("./index");

describe("EPATH", () => {
    describe("DATA Segment Build Utility", () => {
        it("Generates Appropriate Output", () => {
            let test = build("TotalCount");
            expect(test).toMatchSnapshot();

            test = build(Buffer.from([0x1001, 0x2002, 0x3003]), false);
            expect(test).toMatchSnapshot();

            test = build("SomeTag"); // test symbolic build
            expect(test).toMatchSnapshot();

            test = build("0"); // test element build
            expect(test).toMatchSnapshot();

            test = build("255"); // test 8bit upper boundary
            expect(test).toMatchSnapshot();

            test = build("256"); // test 16 bit lower boundary
            expect(test).toMatchSnapshot();

            test = build("257"); // test 16 bit endian
            expect(test).toMatchSnapshot();

            test = build("65535"); // test 16 bit upper boundary
            expect(test).toMatchSnapshot();

            test = build("65536"); // test 32 bit lower boundary
            expect(test).toMatchSnapshot();

            test = build("65537"); // test 32 bit endian
            expect(test).toMatchSnapshot();
        });

        it("Throws with Bad Input", () => {
            const fn = (data, ansi = true) => {
                return () => {
                    build(data, ansi);
                };
            };

            expect(fn("hello")).not.toThrow();
            expect(fn(32)).toThrow();
            expect(fn({ prop: 76 })).toThrow();
            expect(fn(Buffer.from("hello world"))).not.toThrow();
            expect(fn(Buffer.from("hello world"), false)).not.toThrow();
            expect(fn(1, -1)).toThrow();
            expect(fn(1, { hey: "you" })).toThrow();
        });
    });
});
