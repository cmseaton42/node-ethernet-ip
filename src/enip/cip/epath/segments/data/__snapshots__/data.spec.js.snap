// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 1`] = `
Object {
  "data": Array [
    145,
    10,
    84,
    111,
    116,
    97,
    108,
    67,
    111,
    117,
    110,
    116,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 2`] = `
Object {
  "data": Array [
    128,
    2,
    1,
    2,
    3,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 3`] = `
Object {
  "data": Array [
    145,
    7,
    83,
    111,
    109,
    101,
    84,
    97,
    103,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 4`] = `
Object {
  "data": Array [
    40,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 5`] = `
Object {
  "data": Array [
    40,
    255,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 6`] = `
Object {
  "data": Array [
    41,
    0,
    0,
    1,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 7`] = `
Object {
  "data": Array [
    41,
    0,
    1,
    1,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 8`] = `
Object {
  "data": Array [
    41,
    0,
    255,
    255,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 9`] = `
Object {
  "data": Array [
    42,
    0,
    0,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`EPATH DATA Segment Build Utility Generates Appropriate Output 10`] = `
Object {
  "data": Array [
    42,
    0,
    1,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;
